src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// DiagnosticHandler notifies DiagnosticSource subscribers about outgoing Http requests
+    /// </summary>
+    internal sealed class DiagnosticsHandler : DelegatingHandler
+    {
+        /// <summary>
+        /// DiagnosticHandler constructor
+        /// </summary>
+        /// <param name="innerHandler">Inner handler: Windows or Unix implementation of HttpMessageHandler. 
+        /// Note that DiagnosticHandler is the latest in the pipeline </param>
+        public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
+        {
+        }
+
+        internal static bool IsEnabled()
+        {
+            return s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ResponseWriteName);
```

Comment:
@avanderhoorn I'm working on [Activity](https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.DiagnosticSource/src/ActivityUserGuide.md) (new class in System.Diagnostics.DiagnosticSource) support in HttpClient.
As a part of this work, two new events will be fired: System.Net.Http.Activity.Start and Stop.
They will have other payloads, so we cannot reuse existing Request/Response events.

In the Write callback, users will receive request (response) objects and they also may access `Activity.Current` as a diagnostics context: Activity carries Id, start/stop time and some other properties.

**Once new events are implemented, I suggest to deprecate Request/Response and let users migrate to Activity events instead, since they (along with Activity.Current context) will be more descriptive.**

I'd be happy to document **new** events (e.g. as a .md file in this repo) and recommend to use them.
Would you be ok with this?